### PR TITLE
Qr Code Fix | new Camera Package Expo51

### PR DESCRIPTION
**Pull request by Harshksaw**
Screenshot 2024-05-22 at 7 30 18 PM
WhatsApp Image 2024-05-22 at 19 31 53 (1)
WhatsApp Image 2024-05-22 at 19 31 54

### CODE REVIEW

File: .vscode/settings.json
Comment (Bhaumik-Tandan) [inline .vscode/settings.json:1]: please remove this file and add it to .gitignore it is your own local setup

File: helper/linking.js [hunk 1]
```diff
@@ -12,10 +12,13 @@ const linking = {
...
-            Linking.removeEventListener('url', onReceiveURL);
-        };
+
+    const subscription = Linking.addEventListener('url', onReceiveURL);
+
+    return () => {
+
+        subscription.remove();
+    };
     },
```
Comment (Bhaumik-Tandan) [inline helper/linking.js:22]: why do we need to change this file?

**Review by Bhaumik-Tandan — CHANGES_REQUESTED** [review]

File: helper/linking.js (hunk 1 above)
Comment (Harshksaw) [reply helper/linking.js:22]: In Expo SDK 51, the Linking API has been deprecated and replaced with the Linking API from expo-linking package. The addEventListener method is no longer available. Instead, should use the addLinkingListener method.

File: components/CameraScanner.js [hunk 2]
```diff
@@ -21,28 +20,76 @@ const CameraScanner = ({ handleBarCodeScanned, isLit, setIsLit }) => {
         }
         alert('No QR code found in image');
     }
+    const [scanned, setScanned] = useState(false);
+
+    useEffect(() => {
+        const fun = async () => {
+            const checkcamera = await BarCodeScanner.getPermissionsAsync();
+            console.log(checkcamera);
```
[the remark this answers is not in the document]
Comment (Harshksaw) [reply components/CameraScanner.js:28]: this is local state checking again and again , not the permission

File: components/CameraScanner.js [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+import React, { useCallback, useEffect, useState } from 'react';
+import { View, StyleSheet, Text, TouchableOpacity, Button } from 'react-native';
 import QRIndicator from './QRIndicator';
 import QRFooterButton from './QRFooterButton';
 import { calcHeight, calcWidth } from '../helper/res';
 import * as BarCodeScanner from 'expo-barcode-scanner';
 import getLocalImage from '../helper/getLocalImage';
 import getQrDataFromImage from '../helper/getQrDataFromImage';
-
+import { CameraView, useCameraPermissions } from 'expo-camera';
```
Comment (Bhaumik-Tandan) [inline components/CameraScanner.js:9]: what is the use of useCameria permissions here

File: .vscode/settings.json
Comment (Bhaumik-Tandan) [inline .vscode/settings.json:1]: the file is still not ignored

File: components/CameraScanner.js [hunk 4]
```diff
@@ -21,28 +20,70 @@ const CameraScanner = ({ handleBarCodeScanned, isLit, setIsLit }) => {
...
-                onBarCodeScanned={handleBarCodeScanned}
-                style={StyleSheet.absoluteFill}
-                flashMode={isLit ? FlashMode.torch : FlashMode.off}
-            />
-            <QRIndicator />
-            <View style={[styles.footer, { bottom: 30 + bottom }]}>
-                <QRFooterButton onPress={() => setIsLit((isLit) => !isLit)} isActive={isLit} iconName="flashlight-sharp" />
-                <QRFooterButton onPress={getImage} iconName="image" />
-            </View>
+                onBarcodeScanned={scanned ? console.log('not scanned') : handleBarCode}
```
Comment (Bhaumik-Tandan) [inline components/CameraScanner.js:45]: what is the use of console.log log here
why do we have the scanned state?

**Review by Bhaumik-Tandan — CHANGES_REQUESTED** [review]

**Review by Bhaumik-Tandan — APPROVED** [review]

**Pull request merged**